Code/Framework/AzToolsFramework/AzToolsFramework/ComponentModes/ViewportEditUtilities.cpp

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzToolsFramework/ComponentModes/ViewportEditUtilities.h>
+
+namespace AzToolsFramework
+{
+    template <typename T>
+    AZ::Vector3 GetPositionInManipulatorFrame(float worldUniformScale, const AZ::Transform& manipulatorLocalTransform,
+        const T& action)
+    {
+        return manipulatorLocalTransform.GetInverse().TransformPoint(
+            action.m_start.m_localPosition +
+            action.LocalPositionOffset() / AZ::GetClamp(worldUniformScale, AZ::MinTransformScale, AZ::MaxTransformScale));
+    }
```

Comment:
question: Should this template function be in the .cpp? 🤔 I think this could cause problems unless you're using explicit instantiations somewhere, otherwise it should be in the header